site/src/pages/tutorials/grpc/blog/add-service-to-server.mdx

```diff
@@ -0,0 +1,187 @@
+                GrpcService.builder()
+                           .addService(new BlogService())
+                           .enableUnframedRequests(true)
+                           .build();
+}
+```
+
+Map the exceptions our blog service throws to grPC status by calling `exceptionMapping()` with the exception handler we defined.
+
+```java filename=Main.java
+static Server newServer(int port) {
+  final GrpcService grpcService =
+                GrpcService.builder()
+                           .addService(new BlogService())
+                           .enableUnframedRequests(true)
+                           .exceptionMapping(new GrpcExceptionHandler())
+                           .build();
+}
+```
+
+## 3. Add Doc service
+
+To test our service works, add Armeria's [Documentation service](/docs/server-docservice). You may skip this step.
+
+In the `newServer()` method, first create a request example which is displayed in the DocService page. Feel free to add more examples for other service methods.
+
+```java filename=Main.java highlight=1,6-13
+import com.linecorp.armeria.server.docs.DocService;
+
+public final class Main {
+  static Server newServer(int port) {
+    final BlogPost exampleRequest = BlogPost.newBuilder()
+                                            .setTitle("My first blog")
+                                            .setContent("Hello Armeria!")
+                                            .build();
+
+    final DocService docService = DocService.builder()
+                                            .exampleRequests(BlogServiceGrpc.SERVICE_NAME,
+                                                            "CreateBlogPost", exampleRequest)
+                                            .exclude(DocServiceFilter.ofServiceName(
+                                                    ServerReflectionGrpc.SERVICE_NAME))
+                                            .build();
+  }
+}
+```
+
+To easily access the DocService page, add a log message at the end of the `main()` method.
+
+```java filename=Main.java highlight=2
+public static void main(String[] args) throws Exception {
+  ...
+  logger.info("Server has been started. Serving DocService at http://127.0.0.1:{}/docs",
+                server.activeLocalPort());
+}
+```
+
+## 4. Add services to server
+
+As the last part of the `newServer()` method, add the gRPC service and DocService to the server.
+
+```java filename=Main.java
+static Server newServer(int port) {
+  ...
+  return Server.builder()
+               .http(port)
+               .service(grpcService)
+               .serviceUnder("/docs", docService) // Optional
+               .build();
+}
+```
+
+## 5. Run service
+
+To run the service, build the service first.
+// TW question: Is this right?
+
+```bash
+./gradlew build
+```
+
+Run the service by running the `main()` method or using Gradle.
+
+// TW: This should work right..? Mine doesn't work if I run this on a terminal, but it does if I run it from IDE.
```

Comment:
What error message do you get?

Reply:
(scratching my head... ) I don't recall the message at all. Ah, but , Ikhoon helped me out on this :) Thanks for checking this out.